creator-node/src/services/stateMachineManager/monitoring/processStateMonitoringJob.js

```diff
@@ -0,0 +1,276 @@
+const config = require('../../../config')
+const { logger } = require('../../../logging')
+const NodeHealthManager = require('../nodeHealthManager')
+const NodeToSpIdManager = require('../nodeToSpIdManager')
+const {
+  getNodeUsers,
+  buildReplicaSetNodesToUserWalletsMap,
+  computeUserSecondarySyncSuccessRatesMap,
+  aggregateReconfigAndPotentialSyncOps
+} = require('./utils')
+const {
+  retrieveClockStatusesForUsersAcrossReplicaSet
+} = require('../clockUtils')
+
+// Number of users to process each time processStateMonitoringJob is called
+const USERS_PER_JOB = config.get('snapbackUsersPerJob')
+const THIS_CNODE_ENDPOINT = config.get('creatorNodeEndpoint')
+
+/**
+ * Processes a job to monitor the current state of `USERS_PER_JOB` users.
+ * Returns the syncs and replica set updates that are required for these users.
+ * @param {number} jobId the id of the job being run
+ * @param {number} lastProcessedUserId the highest ID of the user that was most recently processed
+ * @param {string} discoveryNodeEndpoint the IP address / URL of a Discovery Node to make requests to
+ * @param {number} moduloBase (DEPRECATED)
+ * @param {number} currentModuloSlice (DEPRECATED)
+ * @return {Object} { lastProcessedUserId (number), jobFailed (boolean) }
+ */
+const processStateMonitoringJob = async (
+  jobId,
+  lastProcessedUserId,
+  discoveryNodeEndpoint,
+  moduloBase, // TODO: Remove. https://linear.app/audius/issue/CON-146/clean-up-modulo-slicing-after-all-dns-update-to-support-pagination
+  currentModuloSlice // TODO: Remove. https://linear.app/audius/issue/CON-146/clean-up-modulo-slicing-after-all-dns-update-to-support-pagination
+) => {
+  // Record all stages of this function along with associated information for use in logging
```

Comment:
decision tree steps are the same as the [steps from snapback up until issueReplicaSetUpdateOps](https://github.com/AudiusProject/audius-protocol/blob/4c559fd7b2a7e3aa5e4772c5b153260841fe9072/creator-node/src/snapbackSM/snapbackSM.js#L969-L1163). the main difference is that the `updateEndpointToSpIdMap` step is removed (see above comment on why)